lib/utils/fs.go

```diff
@@ -209,3 +210,46 @@ func FSTryReadLockTimeout(ctx context.Context, filePath string, timeout time.Dur
 
 	return unlockWrapper(fileLock.Unlock, fileLock.Path()), nil
 }
+
+// RemoveSecure attempts to securely delete the file by first overwriting the file with random data followed by
+// calling os.Remove(filePath). The number of times to overwrite the data is set by iterations. Iterations must
+// be > 0 or a trace.BadParameterError will be returned.
+func RemoveSecure(filePath string, iterations int) error {
+	if iterations < 1 {
+		return trace.BadParameter("iterations must be greater than 0")
+	}
+
+	for i := 0; i < iterations; i++ {
+		if err := overwriteFile(filePath); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+	return trace.Wrap(os.Remove(filePath))
+}
+
+func overwriteFile(filePath string) error {
+	f, err := os.OpenFile(filePath, os.O_WRONLY, 0)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer f.Close()
```

Comment:
I would check the close error, since that's when the buffer is flushed.

Reply:
+1